toonz/sources/toonz/mainwindow.cpp

```diff
@@ -1418,14 +1418,23 @@ QAction *MainWindow::createAction(const char *id, const QString &name,
   QAction *action = new DVAction(name, this);
   addAction(action);
 #ifdef MACOSX
+// To prevent the wrong menu items (due to MacOS menu naming conventions), from
+// taking Preferences, Quit, or About roles (sometimes happens unexpectedly in
+// translations) - all menu items should have "NoRole"
+  if (MenuFileCommandType || MenuEditCommandType || MenuScanCleanupCommandType
+      || MenuLevelCommandType || MenuXsheetCommandType || MenuCellsCommandType || 
+      MenuViewCommandType || MenuWindowsCommandType) {
```

Comment:
Thank you @artisteacher for tackling with the OSX issue !

It seems that this if statement condition will always be TRUE, since you don't compare any values and the type values listed here are all larger than zero.

How about rewriting lines 1424-1437 like as follows? : 

```
  if (strcmp(id, MI_Preferences) == 0)
    action->setMenuRole(QAction::PreferencesRole);
  else if (strcmp(id, MI_Quit) == 0)
    action->setMenuRole(QAction::QuitRole);
  else if (strcmp(id, MI_About) == 0)
    action->setMenuRole(QAction::AboutRole);
  else
    action->setMenuRole(QAction::NoRole);
```